core/src/main/java/org/bitcoinj/core/ECKey.java

```diff
@@ -186,16 +191,24 @@ protected ECKey(@Nullable BigInteger priv, ECPoint pub) {
      * Utility for compressing an elliptic curve point. Returns the same point if it's already compressed.
      * See the ECKey class docs for a discussion of point compression.
      */
-    public static ECPoint compressPoint(ECPoint uncompressed) {
-        return CURVE.getCurve().decodePoint(uncompressed.getEncoded(true));
+    public static ECPoint compressPoint(ECPoint point) {
+        return point.isCompressed() ? point : CURVE.getCurve().decodePoint(point.getEncoded(true));
```

Comment:
I'm not sure if this will further help performance, but instead of encoding/decoding in these functions we could call `Curve.createPoint(point.getX(), point.getY(), true)`, etc. .  After all, compression is just a flag, and doesn't really require any math.